### PR TITLE
fix(query-generator): syntax error in Sqlite with TINYINT and AUTOINCREMENT primary key

### LINKED ISSUE #9995: Sqlite dialect ERROR on AUTOINCREMENT id with datatype different then INTEGER.
## What are you doing?

The following model definition raises an error when using Sqlite dialect:

```
sequelize.define('User', {
        id: {
            type: DataTypes.SMALLINT,
            allowNull: false,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING(25),
            allowNull: false,
        }, {});
```
while this is working properly: 
```
...       
    id: {
            type: DataTypes.INTEGER(4),
            allowNull: false,
            primaryKey: true,
            autoIncrement: true
        },
...
```
## What do you expect to happen?
It should work both with **INTEGER(length)** and **SMALLINT**.

## What is actually happening?
I got the following error:

    'SQLITE_ERROR: AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY'

__Dialect:__ mysql (production)/ sqlite (development)
__Dialect version:__ 4.0.2
__Sequelize version:__ 4.39.0

### PR DESCRIPTION
**Pull request by ugogiordano87**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Fixes #9995

### CODE REVIEW

**Comment by ugogiordano87** [comment]
The first rule of the [column affinity rules](https://www.sqlite.org/datatype3.html#determination_of_column_affinity) specified in the SQLite documentation says:
      
> If the declared type contains the string "INT" then it is assigned INTEGER affinity.

So SQLite ignores the **UNSIGNED** keyword when converting the datatypes into affinities. 

Only the **ZEROFILL** option does't seem to be supported by SQLite.

**Comment by sushantdhiman** [comment]
Is `DATATYPE UNSIGNED` actually an unsigned column? I think for SQLite it will accept any integer values you throw at it.

So `INT UNSIGNED` should accept `1` but rejected `-1`, but I think SQLite accepts both. If that is the case we can just remove UNSIGNED option for SQLite

**Comment by ugogiordano87** [comment]
Well you're right, it seems that SQLite accept both. So we can remove the UNSIGNED options, maybe with a simple function like this:

```
function removeUnsupportedOptions(dataType) {
    if (dataType._zerofill || dataType._unsigned) {
      warn('SQLite does not support ' + dataType._name + '  with UNSIGNED or ZEROFILL. Plain `' + dataType._name + '` will be used instead.');
      dataType._unsigned = undefined;
      dataType._zerofill = undefined;
    }
  }
```

called inside each INT datatype.

**Comment by sushantdhiman** [comment]
@ugogiordano87 Looks good

File: lib/dialects/postgres/data-types.js [hunk 1]
```diff
@@ -7,6 +7,20 @@ const inherits = require('../../utils/inherits');
 module.exports = BaseTypes => {
   const warn = BaseTypes.ABSTRACT.warn.bind(undefined, 'http://www.postgresql.org/docs/9.4/static/datatype.html');
 
+  /**
+   * Removes unsupported Postgres options, i.e., LENGTH, UNSIGNED and ZEROFILL, for the integer data types.
+   * @param dataType The base integer data type.
+   */
+  function removeUnsupportedOptions(dataType) {
+    if (this._length || this.options.length || this._unsigned || this._zerofill) {
+      warn('SQLite does not support ' + dataType._name + '  with LENGTH, UNSIGNED or ZEROFILL. Plain `' + dataType._name + '` will be used instead.');
```
Comment (sushantdhiman) [inline lib/dialects/postgres/data-types.js]: `PostgreSQL does not support <typename> with options. Plain '<typename>' will be used instead.`

Comment (ugogiordano87) [reply lib/dialects/postgres/data-types.js]: Yes I see. What do you think if I define an util file like this:

```
'use strict';

/**
 * Removes unsupported options, i.e., LENGTH, UNSIGNED and ZEROFILL, for the integer data types.
 *
 * @param dialect The specific SQL dialect
 * @param dataType The base integer data type.
 * @param warn The dialect specific warn function.
 */
function removeUnsupportedOptions(dialect, dataType, warn) {
  let warnMessage = '';

  if (this._length || this.options.length || this._unsigned || this._zerofill) {
    this._unsigned = undefined;
    this._zerofill = undefined;

    if (dialect === 'sqlite') {
      warnMessage = 'SQLite does not support ' + dataType._name + '  with UNSIGNED or ZEROFILL. ';
    } else if (dialect === 'postgres') {
      warnMessage = 'PostgresSQL does not support ' + dataType._name + '  with LENGTH, UNSIGNED or ZEROFILL. ';
      this._length = undefined;
      this.options.length = undefined;
    }
    
    warn(warnMessage + 'Plain `' + dataType._name + '` will be used instead.');
  }
}


module.exports = removeUnsupportedOptions;
```

Comment (sushantdhiman) [reply lib/dialects/postgres/data-types.js]: No that will overtime become complex to handle different dialect and their particularities. Keep these functions in `data-types.js` of those dialects, so they can handle dialect specific changes

Also rename `removeUnsupportedOptions` => `removeUnsupportedIntegerOptions`

Comment (ugogiordano87) [reply lib/dialects/postgres/data-types.js]: Sorry, dataType instead of this.

Comment (ugogiordano87) [reply lib/dialects/postgres/data-types.js]: Done.

**Comment by sushantdhiman** [comment]
Thanks @ugogiordano87

**Comment by ugogiordano87** [comment]
You're welcome @sushantdhiman

**Pull request merged**